src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -1021,127 +1011,39 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
 
   public async simulateTransaction(
     transaction: SimulationTransaction,
-    parentBlock: Block,
+    simulationBlock: Block,
     overrides: CallOverrides
   ) {
-    let result: EVMResult;
-
-    const data = transaction.data;
-    let gasLimit = transaction.gas.toBigInt();
-    // subtract out the transaction's base fee from the gas limit before
-    // simulating the tx, because `runCall` doesn't account for raw gas costs.
-    const hasToAddress = transaction.to != null;
-    let to: EthereumJsAddress = null;
-    if (hasToAddress) {
-      const toBuf = transaction.to.toBuffer();
-      to = {
-        equals: (a: { buf: Buffer }) => toBuf.equals(a.buf),
-        buf: toBuf
-      } as any;
-    } else {
-      to = null;
-    }
-
     const common = this.fallback
       ? this.fallback.getCommonForBlockNumber(
           this.common,
           BigInt(transaction.block.header.number.toString())
         )
       : this.common;
+    const simHandler = new SimulationHandler(
+      this,
+      common,
+      true,
+      this.#emitStepEvent
+    );
+    await simHandler.initialize(simulationBlock, transaction, overrides);
+    const callResult = await simHandler.runCall();
+    return Data.from(callResult.execResult.returnValue || "0x");
```

Comment:
We could use `Data.Empty` constant here:
```suggestion
    const callResultValue = callResult.execResult.returnValue;
    return callResultValue === undefined ? Data.Empty : Data.from(callResultValue);
```

I don't feel especially strongly on this, but it might be worthwhile so that we don't have to re-create the empty `Data`

Reply:
Done!